src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -333,6 +335,64 @@ public void noPropertiesWarnings() throws Exception {
     }
 
+    @Issue("JENKINS-37219")
+    @Test public void disableTriggers() throws Exception {
+        sampleRepo.init();
+        sampleRepo.write("Jenkinsfile", "properties([overrideIndexTriggers(false)])");
+        sampleRepo.git("add", "Jenkinsfile");
+        sampleRepo.git("commit", "--message=init");
+
+        WorkflowMultiBranchProject p = r.jenkins.createProject(WorkflowMultiBranchProject.class, "p");
+        BranchSource branchSource = new BranchSource(new GitSCMSource("source-id", sampleRepo.toString(), "", "*", "", false));
+        p.getSourcesList().add(branchSource);
+
+        // Should be initial build of master, which sets the job property.
+        WorkflowJob master = WorkflowMultiBranchProjectTest.scheduleAndFindBranchProject(p, "master");
+        r.waitUntilNoActivity();
+        r.assertBuildStatusSuccess(master.getBuildByNumber(1));
+        assertEquals(2, master.getNextBuildNumber());
+
+        assertNotNull(master.getProperty(OverrideIndexTriggersJobProperty.class));
+        assertFalse(master.getProperty(OverrideIndexTriggersJobProperty.class).getEnableTriggers());
+
+        sampleRepo.git("checkout", "master");
+        sampleRepo.write("Jenkinsfile", "properties([])");
+        sampleRepo.git("commit", "--all", "--message=master-2");
+        sampleRepo.notifyCommit(r);
+
+        WorkflowMultiBranchProjectTest.showIndexing(p);
+        // Should not be a new build.
+        assertEquals(2, master.getNextBuildNumber());
+
+        // Should be able to manually build master, which should result in the blocking going away.
+        WorkflowRun secondBuild = r.assertBuildStatusSuccess(master.scheduleBuild2(0));
+        assertNotNull(secondBuild);
+        assertEquals(2, secondBuild.getNumber());
+        assertEquals(3, master.getNextBuildNumber());
+        assertNull(master.getProperty(OverrideIndexTriggersJobProperty.class));
+
+
+        // Now let's see it actually trigger another build from a new commit.
+        sampleRepo.git("checkout", "master");
+        sampleRepo.write("Jenkinsfile", "// yet more");
+        sampleRepo.git("commit", "--all", "--message=master-3");
+        sampleRepo.notifyCommit(r);
+        WorkflowMultiBranchProjectTest.showIndexing(p);
+        assertEquals(4, master.getNextBuildNumber());
+    }
+
+    @Issue("JENKINS-37219")
+    @Test
+    public void snippetGeneratorOverrideIndexing() throws Exception {
+        String snippetJson = "{'propertiesMap':\n" +
+                "{'stapler-class-bag': 'true', 'jenkins-branch-OverrideIndexTriggersJobProperty': \n" +
+                "{'specified': true, 'enableTriggers': true}},\n" +
+                "'stapler-class': 'org.jenkinsci.plugins.workflow.multibranch.JobPropertyStep',\n" +
+                "'$class': 'org.jenkinsci.plugins.workflow.multibranch.JobPropertyStep'}";
+
+        new SnippetizerTester(r).assertGenerateSnippet(snippetJson, "properties [overrideIndexTriggers(true)]", null);
```

Comment:
Oh, yeah, this isn't testing `isOwnerMultibranch` at *all* because OW OW OW OW. It's just testing that the generation actually does what it's supposed to. Or not. Y'know.